app/src/main/java/org/freedombox/freedombox/views/fragments/DiscoveryFragment.kt

```diff
@@ -18,18 +18,37 @@
 package org.freedombox.freedombox.views.fragments
 
 import android.content.Context.NSD_SERVICE
+import android.content.Intent
 import android.content.SharedPreferences
 import android.net.nsd.NsdManager
 import android.net.nsd.NsdServiceInfo
 import android.os.Bundle
+import android.preference.PreferenceManager
 import android.util.Log
+import android.view.View
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
+import kotlinx.android.synthetic.main.fragment_discovery.*
 import org.freedombox.freedombox.R
 import org.freedombox.freedombox.components.AppComponent
+import org.freedombox.freedombox.views.activities.LauncherActivity
+import org.freedombox.freedombox.views.adapter.DiscoveryListAdapter
+import org.freedombox.freedombox.views.model.ConfigModel
 import javax.inject.Inject
 
 class DiscoveryFragment : BaseFragment() {
     val TAG = "DISCOVERY_FRAGMENT"
 
+    lateinit var adapter: DiscoveryListAdapter
+
+    var discoveredBoxList: ArrayList<String> = ArrayList<String>()
+    var configuredBoxList: ArrayList<String> = ArrayList<String>()
+
+    var disoveredPortList: ArrayList<String> = ArrayList<String>()
```

Comment:
Kotlin has type inference. You needn't explicitly declare types, but it's okay even if you did.

Reply:
Now I have changed it to  val `discoveredBoxList = mutableListOf<String>()`